### ISSUE TITLE
no error when importing something that is conditionally defined

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#16938]
```py
# foo.py
if bool():
    a = 1
```
```py
# bar.py
from foo import a # no error, crashes at runtime
```

i would expect an error like `Name "a" may be undefined`, which is what happens when you try to use it in the same file:
```py

# foo.py
if bool():
    a = 1
print(a) # error: Name "a" may be undefined
```

**Comment by hauntsaninja** [comment]
mypy will detect this with `--enable-error-code possibly-undefined`

**Comment by DetachHead** [comment]
@hauntsaninja i have that rule enabled. the problem is that it doesn't detect it if you're importing a possibly undefined variable from another file, like in my example.

**Comment by hauntsaninja** [comment]
Oh sorry, I misread!
